src/main/java/org/mskcc/cmo/metadb/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -0,0 +1,268 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ValidRequestCheckerImpl implements ValidRequestChecker {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private boolean isCmoRequest;
+    private boolean hasRequestId;
+
+    @Autowired
+    private RequestStatusLogger requestStatusLogger;
+
+    @Value("${igo.cmo_request_filter:false}")
+    private Boolean igoCmoRequestFilter;
+
+    private static final Log LOG = LogFactory.getLog(ValidRequestCheckerImpl.class);
+
+    /**
+     * Checks is the request is a cmoRequest and has a requestId
+     * Sample list is extracted from the requestJsonMap
+     * If the total number of samples available is equal to valid samples,
+     * then the request as a whole is valid
+     * If the number of valid samples is zero, the request is invalid and will be logged
+     * If the number of valid samples is less than the total number of samples, the
+     * request is still considered valid but logged to keep note of the invalid samples
+     * @throws IOException
+     */
+    @Override
+    public String checkIfValidRequest(String requestJson) throws IOException {
+
+        if (StringUtils.isAllBlank(requestJson)) {
+            return null;
+        }
+
+        Map<String, Object> requestJsonMap = mapper.readValue(requestJson, Map.class);
+        isCmoRequest = isCmoRequest(requestJsonMap);
+        hasRequestId = hasRequestId(requestJsonMap);
+
+        if (igoCmoRequestFilter && !isCmoRequest) {
+            LOG.info("CMO request filter enabled - skipping non-CMO request: "
+                    + requestJsonMap.get("requestId"));
+            return null;
+        }
+
+        Object[] sampleList = mapper.convertValue(requestJsonMap.get("samples"),
+                Object[].class);
+        List<Object> validSampleList = new ArrayList<Object>();
+        int numOfSamples = sampleList.length;
+
+        for (Object sample: sampleList) {
+            Map<String, String> sampleMap = mapper.convertValue(sample, Map.class);
+            if (!igoCmoRequestFilter && !isCmoRequest) {
+                if (isValidNonCmoSample(sampleMap)) {
+                    validSampleList.add(sample);
+                }
+            } else {
+                if (isValidCmoSample(sampleMap)) {
+                    validSampleList.add(sample);
+                }
+            }
+
+        }
+
+        if (validSampleList.size() > 0) {
+            if (validSampleList.size() < numOfSamples) {
+                String newSamplesString = mapper.writeValueAsString(validSampleList.toArray(new Object[0]));
+                requestJsonMap.replace("samples", newSamplesString);
+                LOG.info("CMO request passed sanity checking - missing samples...");
+                requestStatusLogger.logRequestStatus(requestJson,
+                        RequestStatusLogger.StatusType.REQUEST_WITH_MISSING_SAMPLES);
```

Comment:
`REQUEST_WITH_MISSING_SAMPLES` is the status we use for a request json that comes with no samples

there's a different status that indicates that a request json has at least one valid sample but some were filtered out